File: src/market-tracker-webapi/Application/Domain/Schemas/Market/Inventory/Product/Product.cs
namespace market_tracker_webapi.Application.Domain.Schemas.Market.Inventory.Product;

public record Product(
    ProductId Id,
    string Name,
    string ImageUrl,
    int Quantity,
    ProductUnit Unit,
    int Views,
    double Rating,
    Brand Brand,
    Category Category
)
{
    public Product(
        string Id,
        string Name,
        string ImageUrl,
        int Quantity,
        ProductUnit Unit,
        int Views,
        double Rating,
        Brand Brand,
        Category Category
    ) : this(
        new ProductId(Id),
        Name,
        ImageUrl,
        Quantity,
        Unit,
        Views,
        Rating,
        Brand,
        Category
    )
    {
    }
}

public record ProductId(string Value);

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/model/market/inventory/product/Product.kt
package pt.isel.markettracker.domain.model.market.inventory.product

import pt.isel.markettracker.domain.model.market.inventory.Brand
import pt.isel.markettracker.domain.model.market.inventory.Category

data class Product(
    val id: String,
    val name: String,
    val imageUrl: String,
    val quantity: Int,
    val unit: ProductUnit,
    val rating: Double,
    val brand: Brand,
    val category: Category
)